plugin/LaserDevice.h

```diff
@@ -73,6 +73,27 @@ namespace smode
           smode::laser::frame_stream_set_radians_per_point(&frame_stream, (float)anglePerPoint);
         }
       }
+
+      if (!isApplyingVariableConstraints()) {
+        if (variable == &dacPointsPerSecond) {
+          VariableConstraintsScope _(*this);
+          uint32_t runtimeMax = dac.kind.ether_dream.broadcast.max_point_rate;
+          uint32_t runtimeMin = 2000;
+          DBG("PPS runtime max = " + String(runtimeMax));
+          if ((uint32_t)dacPointsPerSecond < runtimeMin)
+            dacPointsPerSecond.set(runtimeMin);
+          if ((uint32_t)dacPointsPerSecond > runtimeMax)
+            dacPointsPerSecond.set(runtimeMax);
+        }
+        else if (variable == &latencyPoints) {
+          VariableConstraintsScope _(*this);
+          uint32_t runtimeMax = dac.kind.ether_dream.broadcast.buffer_capacity;
+          DBG("latency runtime max = " + String(runtimeMax));
+          if ((uint32_t)latencyPoints > runtimeMax)
+            latencyPoints.set(runtimeMax);
+        }
+      }
+
       ControlDevice::variableChangedCallback(variable, changedObject);
```

Comment:
I advice doing a `typedef ControlDevice BaseClass; ` (in private section at the bottom of the class definition) and then use:
`BaseClass::variableChangedCallback(variable, changedObject);`

by the way, just a reminder also to extract an abstract class LaserDevice and a concrete class such as NanouLaserDevice